recipes/hdf5/all/conanfile.py

```diff
@@ -67,6 +68,11 @@ def configure(self):
              self.options.szip_encoding and \
              not self.options["szip"].enable_encoding:
             raise ConanInvalidConfiguration("encoding must be enabled in szip dependency (szip:enable_encoding=True)")
+        if self.options.parallel:
+            if self.options.enable_cxx:
+                raise ConanInvalidConfiguration("Parallel and C++ options are mutually exclusive")
+            if self.options.get_safe("threadsafe", False):
+                raise ConanInvalidConfiguration("Parallel and Threadsafe options are mutually exclusive")
```

Comment:
```suggestion
        if self.options.parallel:
            if self.options.enable_cxx:
                raise ConanInvalidConfiguration("Parallel and C++ options are mutually exclusive")
            if self.options.get_safe("threadsafe", False):
                raise ConanInvalidConfiguration("Parallel and Threadsafe options are mutually exclusive")
```

Please, move it to `validate()` method